src/wallet/test/coinselector_tests.cpp

```diff
@@ -917,5 +917,176 @@ BOOST_AUTO_TEST_CASE(effective_value_test)
     BOOST_CHECK_EQUAL(output5.GetEffectiveValue(), nValue); // The effective value should be equal to the absolute value if input_bytes is -1
 }
 
+BOOST_AUTO_TEST_CASE(check_max_weight)
+{
+    const CAmount target = 49.5L * COIN;
+    CCoinControl cc;
+
+    std::unique_ptr<CWallet> tmpWallet = std::make_unique<CWallet>(m_node.chain.get(), "", m_args, CreateMockWalletDatabase());
+    tmpWallet->LoadWallet();
+    LOCK(tmpWallet->cs_wallet);
+    tmpWallet->SetupLegacyScriptPubKeyMan();
+
+    const auto op_dest = tmpWallet->GetNewDestination(OutputType::BECH32, "");
+    BOOST_ASSERT(op_dest);
+    const auto scriptPubKey = GetScriptForDestination(op_dest.value());
+    const auto txout = CTxOut(target, scriptPubKey);
+
+    FastRandomContext rand;
+    CoinSelectionParams cs_params{
+        rand,
+        /*change_output_size=*/34,
+        /*change_spend_size=*/148,
```

Comment:
While these values don't particularly matter, they should probably match the p2wpkh sizes rather than the p2pkh.

```suggestion
        /*change_output_size=*/31,
        /*change_spend_size=*/68,
```